### PR TITLE
LeftJoinCols mod

### PR DESCRIPTION
**Pull request by sallyrobson**

### CODE REVIEW

File: src/types/modifications/AddTableCols.jl [hunk 1]
```diff
@@ -0,0 +1,40 @@
+
+
+struct AddTableCols <: Modification
```
Comment (Ethan-Russell) [inline src/types/modifications/AddTableCols.jl:3]: Be sure to add a doc string before we merge it!  A couple of potential changes:
* Rename the mod to `LeftJoinCols` to be a little more clear about what is happening (or `JoinCols`)
* Rename `join_cols` to `on` to match convention for joining
* Rename `table` to `left_table_name`
* Rename `cols_file` to `right_table_file`
* Rename `cols_table` to `right_table`

**Comment by sallyrobson** [comment]
@Ethan-Russell Thanks for the naming suggestions, does this look better?

**Review by Ethan-Russell — APPROVED** [review]
Looks great! Only remaining question is how it handles missing values? If I remember correctly, left join produces a vector of Union{WhateverType, Missing}, where ever row that doesn't match the "on" cols is filled in with missing. We could provide a missing value, or a missing value for each column that gets added?

**Comment by sallyrobson** [comment]
I added a matchmissing argument, so either you can set it as `notequal` and it will set values that don't match to `missing` or you can set to `equal` and then you would need a `missing` key in the right table with a value to set the missing things to.

**Pull request merged**